app/ide-desktop/lib/dashboard/src/layouts/Settings/DeleteUserAccountSettingsSection.tsx

```diff
@@ -12,15 +11,24 @@ import UnstyledButton from '#/components/UnstyledButton'
 
 import ConfirmDeleteUserModal from '#/modals/ConfirmDeleteUserModal'
 
+import type Backend from '#/services/Backend'
+
 // ========================================
 // === DeleteUserAccountSettingsSection ===
 // ========================================
 
+/** Props for a {@link DeleteUserAccountSettingsSection}. */
+export interface DeleteUserAccountSettingsSectionProps {
+  readonly backend: Backend
```

Comment:
Why do we pass backend instead of consuming it from the `Context`? Props drilling is a bad practice.

Reply:
this is not props drilling, it is one level deep. putting everything in a context to hide state is a bad practice

Reply:
If we need backed deeper inside the React components tree, do we need to pass it as a prop or we can use context instead?

Reply:
typically context (like how it is used everywhere else). i just figured that here specifically the root `Settings` component should be controlling basically all the state of its children

Reply:
So why do we pass the backend directly then?

Reply:
Looks like a stylistic choice that makes no sense and leads to unneeded cognitive load and merge conflicts.
Like a change only to make a change.

Reply:
> unneeded cognitive load 

imo contexts are a higher cognitive load since it is much harder to track where the state is coming from, let alone what its lifecycle looks like (in contrast, with props it's obvious where the state comes from - the parent). if you have a different opinion please do elaborate

> merge conflicts

explain

Reply:
re: merge conflicts specifically - this is something that should never change, so i'm not sure whether this will be a problem (like at all)

also note that the main *reason* why this change is happening is so that the main `Settings` component controls all the state that is passed to each of its children. this means that the inputs are only ever acquired in one place, so that if the inputs change, only the hook in the main `Settings.tsx` needs to be changed, not every such occurrence of the hook in the entire `Settings/` subfolder using a hacky regex search+replace.